internal/ingress/controller/controller.go

```diff
@@ -1090,8 +1090,8 @@ func (n *NGINXController) createServers(data []*ingress.Ingress,
 				servers[host].SSLCiphers = anns.SSLCiphers
 			}
 
-			// only add a certificate if the server does not have one previously configured
-			if servers[host].SSLCert != nil {
+			// only add a certificate if the server does not have one previously configured that is not the default
+			if servers[host].SSLCert != nil && servers[host].SSLCert.PemFileName != n.cfg.DefaultSSLCertificate && servers[host].SSLCert != n.cfg.FakeCertificate {
```

Comment:
@zeeZ I think this is not required. Since I removed the static SSL mode in https://github.com/kubernetes/ingress-nginx/pull/4356 the SSLCert is configured only when the ingress has a valid TLS section, it never contains the default or fake SSL certificate

Reply:
I just checked against the current master and you're absolutely correct, the behavior changed since the last release!